### PR TITLE
quickstart.md: get dependency with git hash instead of version tag

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve? 

Using a version tag to `go get` the dependency is not the correct way. 

```console
go get github.com/pingcap/parser@v4.0.0-rc.1: github.com/pingcap/parser@v4.0.0-rc.1: invalid version: module contains a go.mod file, so major version must be compatible: should be v0 or v1, not v4
```

Under ideal circumstances, the path should be something like 

```
github.com/pingcap/parser/v4@master
``` 

when the new major version is not backward compatible. However, this is still under discussion due to potential compatibility problems.

As a workaround, we can specify the git hash manually.

### What is changed and how it works?
See the changes.

### Check List 

Tests 

N/A

Code changes

N/A

Side effects

N/A

Related changes

N/A

### CODE REVIEW

File: docs/quickstart.md [hunk 1]
```diff
@@ -17,18 +17,18 @@ go mod init colx && touch main.go
 
 ## Import Dependencies
 
-First of all, you need to use `go get` to fetch the dependencies:
+First of all, you need to use `go get` to fetch the dependencies(for v4.0.2, the git hash is `3a18f1e`):
```
Comment (djshow832) [inline docs/quickstart.md:20]: What about give the approach of fetching the git hash?

Comment (tangenta) [reply docs/quickstart.md]: Done.

File: docs/quickstart.md [hunk 2]
```diff
@@ -17,18 +17,18 @@ go mod init colx && touch main.go
 
 ## Import Dependencies
 
-First of all, you need to use `go get` to fetch the dependencies:
+First of all, you need to use `go get` to fetch the dependencies through git hash. The git hashes are available in [release page](https://github.com/pingcap/parser/releases). Take `release-4.0.2` as an example:
```
Comment (kennytm) [inline docs/quickstart.md:20]:
```suggestion
First of all, you need to use `go get` to fetch the dependencies through git hash. The git hashes are available in [release page](https://github.com/pingcap/parser/releases). Take `v4.0.2` as an example:
```

File: docs/quickstart.md [hunk 3]
```diff
@@ -17,18 +17,18 @@ go mod init colx && touch main.go
...
+go get -v github.com/pingcap/parser@3a18f1e
 ```
 
 > **NOTE**
 >
 > You may want to use advanced API on expressions (a kind of AST node), such as numbers, string literals, booleans, nulls, etc. It is strongly recommended to use the `types` package in TiDB repo with the following command:
 >
 > ```bash
-> go get -v github.com/pingcap/tidb/types/parser_driver@v4.0.0-rc.1
+> go get -v github.com/pingcap/tidb/types/parser_driver@328b6d0
```
Comment (kennytm) [inline docs/quickstart.md:31]: should this be consistent with line 23 (3a18f1e)?

Comment (tangenta) [reply docs/quickstart.md:31]: This is TiDB v4.0.2.

https://github.com/pingcap/tidb/releases/tag/v4.0.2

**Pull request merged**